plutus-use-cases/src/Plutus/Contracts/Auction.hs

```diff
@@ -243,10 +242,17 @@ auctionSeller value time = do
 currentState
     :: StateMachineClient AuctionState AuctionInput
     -> Contract AuctionOutput BuyerSchema AuctionError (Maybe HighestBid)
-currentState client = mapError StateMachineContractError (SM.getOnChainState client) >>= \case
-    Just (SM.OnChainState{SM.ocsTxOut=TypedScriptTxOut{tyTxOutData=Ongoing s}}, _) -> do
-        tell $ auctionStateOut $ Ongoing s
-        pure (Just s)
+currentState client = do
+  mOcs <- mapError StateMachineContractError (SM.getOnChainState client)
+  case mOcs of
+    Just (ocs, _) -> do
```

Comment:
Maybe use ViewPatterns here so you don't have to duplicate the failure case?

Reply:
Good idea.